sleap_roots/bases.py

```diff
@@ -11,9 +11,18 @@ def get_bases(pts: np.ndarray) -> np.ndarray:
 
     Returns:
         Array of bases (instances, (x, y)).
+        If there is no root, or the roots don't have bases, an empty array of shape
+        (0,2) is returned.
     """
-    # (instances, 2)
-    base_pts = pts[:, 0]
+    # Check for edge cases where the root is missing or doesn't have bases.
+    if len(pts) == 0 or np.isnan(pts[:, 0].all()):
+        # Shape is (0, 2)
+        base_pts = np.empty((0, 2))
```

Comment:
This case is not being reached in the tests. This is because the conditional is incorrect.

`len(pts) == 0` will never be true because we always have the full number of nodes, regardless of how many are visible since we pad with NaNs.

`np.isnan(pts[:, 0].all())` will never be true because of the order of operations. In the inner expression, you have this: `pts[:, 0].all()` which won't be true since they're numbers, not booleans. This should instead be: `np.isnan(pts[:, 0]).all()` with the `.all()` on the outside.

What is clear from the tests passing, however, is that you don't need this conditional at all since it turns out that you get the expected outputs in all cases with just the `else` clause.

Reply:
Ok, I took out the conditional statement.